tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -203,7 +203,7 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
   /** Desugars parameters and body to a JS function.
    */
   private[emitter] def desugarToFunction(
-      classEmitter: ScalaJSClassEmitter, enclosingClassName: String,
+      classEmitter: ScalaJSClassEmitter, enclosingClassName: Option[String],
```

Comment:
Just to double check. `enclosingClassName` is becoming optional because:

- It makes more sense.
- To prevent static methods of SJS defined JS classes from being treated specially w.r.t. their classes.

If it is not too much of a hassle, consider moving this change into a separate commit.